src/components/PhotoAnalysis/PhotoAnalysis.tsx

```diff
@@ -25,70 +25,95 @@ const Analysis: React.FC = () => {
   >("")
   const [imageSrc, setImageSrc] = useState<string>()
   let labels: string[] = [""]
-  let labelData: Label[]
+  let labelData: LabelType[] = []
   const [isLoading, setIsLoading] = useState(false)
+  const [isError, setError] = useState(false)
 
-  const identifyImageLabels = async (event: any) => {
-    setIsLoading(true)
-
+  const identifyImageResponse = async (event: any) => {
     const files = (event.target as HTMLInputElement).files
     const file = files![0]
-    setImageSrc(URL.createObjectURL(file))
-
-    await Predictions.identify({
-      labels: {
-        source: {
-          file,
+    if (isValidFileType(file)) {
+      setIsLoading(true)
+      setImageSrc(URL.createObjectURL(file))
+      await Predictions.identify({
+        labels: {
+          source: {
+            file,
+          },
+          type: "ALL",
         },
-        type: "ALL",
-      },
-    })
-      .then(response => {
-        setRekognitionResponse(response)
-        setIsLoading(false)
       })
-      .catch(err => setRekognitionResponse(JSON.stringify(err, null, 2)))
+        .then(response => {
+          setRekognitionResponse(response)
+          setIsLoading(false)
+        })
+        .catch(err => setRekognitionResponse(JSON.stringify(err, null, 2)))
+    } else setError(true)
   }
 
   const processRekognitionLabels = (
     rekognitionResponse: IdentifyLabelsOutput
   ) => {
-    let labelsArray: Label[] = []
+    let labelsArray: LabelType[] = []
     if (rekognitionResponse != "") {
-      rekognitionResponse.labels!.forEach(label => {
-        const metadata = label.metadata as Metadata
+      if (rekognitionResponse.labels) {
+        rekognitionResponse.labels!.forEach(label => {
+          const metadata = label.metadata as Metadata
 
-        const { confidence, parents } = metadata
-        if (confidence > 85 && parents.length >= 1) {
-          labelsArray.push(label)
-        }
-      })
+          const { confidence, parents } = metadata
+          if (confidence > 85 && parents.length >= 1) {
+            labelsArray.push(label)
+          }
+        })
+      }
     }
     labelData = labelsArray
     const labelValues = labelsArray.map(label => {
       return label.name
     })
     labels = labelValues
   }
+
+  const displayError = (error: string) => {
```

Comment:
This should be extracted into an Error component I think but for now I just left it as a function as I'm a little unsure of the component design for that